src/librustc_codegen_ssa/back/link.rs

```diff
@@ -968,7 +968,81 @@ pub fn print_native_static_libs(sess: &Session, all_native_libs: &[NativeLibrary
     }
 }
 
+// Because windows-gnu target is meant to be self-contained for pure Rust code it bundles
+// own mingw-w64 libraries. These libraries are often not compatible with mingw-w64
+// installed in the system. This breaks many cases where Rust is mixed with other languages
+// (e.g. *-sys crates).
+// We prefer system mingw-w64 libraries if they are available to avoid this issue.
+fn get_crt_libs_path(sess: &Session) -> Option<PathBuf> {
```

Comment:
Sorry probably should have commented on this earlier, but since `Session` is passed in here can `lazy_static` be avoided? This could presumably be some sort of `Cell` inside of `Session` to cache the value we're interested in